aptos-move/framework/aptos-token/sources/listing.move

```diff
@@ -0,0 +1,243 @@
+module aptos_token::listing {
+    use std::signer;
+    use std::string::String;
+    use aptos_std::table::{Self, Table};
+    use aptos_token::token::{Self, TokenId};
+    use aptos_std::guid::{Self, ID};
+    use aptos_framework::account;
+    use aptos_framework::event::{Self, EventHandle};
+
+
+    const EOWNER_NOT_HAVING_ENOUGH_TOKEN: u64 = 1;
+    const ELISTING_NOT_EXIST:u64 = 2;
+
+    /// immutable struct for recording listing info.
+    struct Listing<phantom CoinType> has drop, store {
+        id: ID,
+        owner: address,
+        token_id: TokenId,
+        amount: u64,
+        min_price: u64,
+        instant_sale: bool, // true for marketplace and false for auction
+        start_sec: u64, // timestamp in secs for the listing starting time
+        expiration_sec: u64, // timestamp in secs for the listing expiration time
+    }
+
+    /// return a listing struct, marketplace owner can use this function to create a listing and store it in its inventory
+    public fun create_list<CoinType>(
+        owner: &signer,
+        token_id: TokenId,
+        amount: u64,
+        min_price: u64,
+        instant_sale: bool,
+        start_sec: u64,
+        expiration_sec: u64
+    ): Listing<CoinType> {
+        let owner_addr = signer::address_of(owner);
+        assert!(token::balance_of(owner_addr, token_id) >= amount, EOWNER_NOT_HAVING_ENOUGH_TOKEN);
+
+        Listing<CoinType> {
+            id: create_listing_id(owner),
+            owner: owner_addr,
+            token_id,
+            amount,
+            min_price,
+            instant_sale,
+            start_sec,
+            expiration_sec,
+        }
+    }
+
+    struct ListingEvent has copy, drop, store {
+        id: ID,
+        token_id: TokenId,
+        amount: u64,
+        min_price: u64,
+        instant_sale: bool,
+        start_sec: u64,
+        expiration_sec: u64,
+        market_address: address,
+    }
+
+    struct CancelListingEvent has copy, drop, store {
+        id: ID,
+        market_address: address,
+    }
+
+    /// store listings on the owner's account
+    struct ListingRecords<phantom CoinType> has key {
+        records: Table<ID, Listing<CoinType>>,
+        listing_event: EventHandle<ListingEvent>,
+        cancel_listing_event: EventHandle<CancelListingEvent>,
+    }
+
+    public fun initialize_listing_records<CoinType>(owner: &signer){
+        let owner_addr = signer::address_of(owner);
+
+        if (!exists<ListingRecords<CoinType>>(owner_addr)) {
+            move_to(
+                owner,
+                ListingRecords<CoinType> {
+                    records: table::new(),
+                    listing_event: account::new_event_handle<ListingEvent>(owner),
+                    cancel_listing_event: account::new_event_handle<CancelListingEvent>(owner),
+                }
+            );
+        };
+    }
+
+    public fun create_list_under_user_account<CoinType>(
```

Comment:
why do we need this versus just the entry function or the list function.

Reply:
This is for 2nd usecase where creator want to directly list under their account, own the listing sell them directly.